core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -5200,27 +5186,65 @@ protected function searchMenu()
 		// Set the search value from the session
 		elseif ((string) $session['search'][$this->strTable]['value'] !== '')
 		{
-			$strPattern = "CAST(%s AS CHAR) REGEXP ?";
+			$searchValue = $session['search'][$this->strTable]['value'];
+			$fld = $session['search'][$this->strTable]['field'];
 
-			if (substr(Config::get('dbCollation'), -3) == '_ci')
+			try
 			{
-				$strPattern = "LOWER(CAST(%s AS CHAR)) REGEXP LOWER(?)";
+				$this->Database->prepare("SELECT '' REGEXP ?")->execute($searchValue);
+			}
+			catch (DriverException $exception)
+			{
+				// Quote search string if it is not a valid regular expression
+				$searchValue = preg_quote($searchValue);
 			}
 
-			$fld = $session['search'][$this->strTable]['field'];
+			$strReplacePrefix = '';
+			$strReplaceSuffix = '';
+
+			// Decode HTML entities to make them searchable
+			if (empty($GLOBALS['TL_DCA'][$this->strTable]['fields'][$fld]['eval']['decodeEntities']))
+			{
+				$arrReplace = array(
+					'&#35;' => '#',
+					'&#60;' => '<',
+					'&#62;' => '>',
+					'&lt;' => '<',
+					'&gt;' => '>',
+					'&#40;' => '(',
+					'&#41;' => ')',
+					'&#92;' => '\\\\',
+					'&#61;' => '=',
+					'&amp;' => '&',
+				);
+
+				$strReplacePrefix = str_repeat('REPLACE(', \count($arrReplace));
+
+				foreach ($arrReplace as $strSource => $strTarget)
+				{
+					$strReplaceSuffix .= ", '$strSource', '$strTarget')";
+				}
+			}
+
+			$strPattern = "$strReplacePrefix CAST(%s AS CHAR) $strReplaceSuffix REGEXP ?";
+
+			if (substr(Config::get('dbCollation'), -3) == '_ci')
+			{
+				$strPattern = "$strReplacePrefix LOWER(CAST(%s AS CHAR)) $strReplaceSuffix REGEXP LOWER(?)";
```

Comment:
I understand that `REPLACE` is case-sensitive according to the docs, but the replaces characters are never upper/lowercase, so what's the `LOWER` for?

Reply:
The `LOWER()` is for the rest of the string in order to compare it case insensitive. This is exactly as before, I didn’t change the `LOWER(CAST(%s AS CHAR))` part.

Reply:
But only if the collation matched? https://github.com/contao/contao/pull/2735/files#diff-e4bf1535e70e3fbc2b9692ba8add4c45237305ff815df14c7ba9c9ce53dc9204L5205

Reply:
Only if the collation was `case insensitive`. Not sure why that would need `LOWER()` at all 😅

Reply:
I would guess that older MySQL versions ignored the collation for `REGEXP` comparisons. But that is not part of this PR IMO.

Reply:
Agreed, but you're now changing the behavior?

Reply:
I don’t understand what you mean, this code is exactly the same as it was before (except the replace prefix and suffix): https://github.com/contao/contao/blob/ffebcd7fa47ce6576d311cf73c2a9eda49ae7113/core-bundle/src/Resources/contao/drivers/DC_Table.php#L5203-L5208

Reply:
Me neither, looking at it now 😂 Sorry for some reason I was blind to see line 5205